src/GogeDao.sol

```diff
@@ -370,718 +476,955 @@ contract GogeDAO is Ownable {
+            polls[_pollNum][msg.sender] = 0;
             totalVotes[_pollNum] -= _numVotes;
-            if (block.timestamp <= pollEndTime[_pollNum]) {
-                historicalTally[_pollNum] -= _numVotes;
-            }
-            require(IGogeERC20(governanceTokenAddr).transfer(_msgSender(), _numVotes));
+
+            _refundVoter(msg.sender, _numVotes);
+            _removeAdvocate(msg.sender, _pollNum);
         }
     }
 
-    /// @notice A method for a voter to remove their votes from all polls.
-    function removeAllVotes() public {
-        for(uint256 i=0; i<=pollNum; i++) {
-            removeVote(i);
+    /// @notice A method for all voters to be refunded after a poll that they've voted on has been passed.
+    /// @param  _pollNum The poll number.
+    function _refundVotersPostChange(uint256 _pollNum) internal {
+        for (uint256 i = 0; i < voterLibrary[_pollNum].length; i++) {
+            address voter  = voterLibrary[_pollNum][i];
+            uint256 amount = polls[_pollNum][voter];
+
+            _refundVoter(voter, amount);
+            _removeAdvocate(voter, _pollNum);
         }
     }
 
+    /// @notice Internal method for transferring governance tokens to a voter.
+    /// @param  _voter address of voter that needs to be refunded.
+    /// @param _amount amount of tokens to refund voter.
+    function _refundVoter(address _voter, uint256 _amount) internal {
+        require(IGogeERC20(governanceTokenAddr).transfer(_voter, _amount));
+    }
 
-    /// @notice A method for a voter to remove their votes from all polls.
-    function removeVotesRange(uint256 startPollNum, uint256 endPollNum) public {
-        for(uint256 i=startPollNum; i<=endPollNum; i++) {
-            removeVote(i);
+    /// @notice A method for removing polls from an address's advocatesFor mapped array.
+    /// @param _advocate address of wallet that we are removing their advocacy.
+    /// @param _pollNum the number of the poll the address is no longer an advocate for.
+    function _removeAdvocate(address _advocate, uint256 _pollNum) internal {
+        uint256 l = advocateFor[_advocate].length;
+        for (uint256 i = 0; i < l; i++) {
+            if (advocateFor[_advocate][i] == _pollNum) {
+                advocateFor[_advocate][i] = advocateFor[_advocate][--l];
+                advocateFor[_advocate].pop();
+            }
         }
     }
-    
-    // ---------- Polls ----------
 
-    /// @notice is used to create a new poll.
-    /// @param  _pollType enum type of poll being created.
-    /// @param  _change the matching metadata that will result in the execution of the poll.
-    function createPoll(PollType _pollType, Metadata memory _change) public {
-        _change.time1 = block.timestamp;
+    /// @notice An internal method for changing gateKeeping status.
+    /// @param _enabled status of gateKeeping.
+    function _setGateKeeping(bool _enabled) internal {
+        require(gateKeeping != _enabled, "Already set");
```

Comment:
Is the require necessary? Also this error message is not exactly helpful. Maybe "Gatekeeping already active"?

Reply:
Well the bool could be false or true